### DISCUSSION TITLE
Activity Hierarchy

### DISCUSSION THREAD

**Discussion by johnsev** [kimai/kimai#3925] [Questions & Answers] [answered]
I've just started working with (evaluating) Kimai Cloud and I'm struggling to understand how to use Activities and Tasks.  I've had my small engineering firm for 30 years.  We break down projects into tasks then track our time by task.  When we create schedules and assign resources, we use the same tasks.  When we quote projects we use the same tasks.

I believe we use Tasks in the way that Kimai uses Activities except that Activities are flat and we don't assign tasks to activities.  For us, a task and activity are the same thing.  So, I'm trying to figure out how we can fit our model to work with Kimai.  Mostly, we use a hierarchical list of activities for a project with WBS codes that follow the hierarchy.  For our projects, we can have anywhere from 3 tasks to a couple hundred.  When we have more than a few tasks, we create a hierarchy that could be many levels deep.  For example, top level task groups of Design, Development, Project Management, and Procurement.  The Design group would have child tasks   Electrical Design, Equipment Design, and Software Design.  Each of those will have child tasks.  Project members can only assign time to tasks that have no child tasks.  The hierarchy depends on how large the project is and how many tasks we have.  We've had as many as 5 levels in the task hierarchy but 2 or 3 is much more common.

Is there any way to do something like this in Kimai - creating Activity hierarchies where users log their time to activities at the bottom of the hierarchy?

**Comment by kevinpapst** [comment, marked as answer]
Thanks for the use-case explanation!

> Is there any way to do something like this in Kimai - creating Activity hierarchies where users log their time to activities at the bottom of the hierarchy?

No it isn't and it won't be added. I can't remember if anyone ever asked for it before, so there is simply not enough need for it.
Quite the contrary: many people want to track times without activities 😁 

No judgement whatsoever, you might have your reasons for the detailed structure. But I never met a customer with such an amount of activities. Maybe you can rethink/simplify the list drastically and use the timesheet description to write down the details?

And then create a flat list of activities:
- Design - Mockup
- Design - Logo
- Design - Figma components ...

**Reply by johnsev** [reply]
It's the nature of the work we do - automating manufacturing plants.  It is how we price and sell a project because someone has to do all these tasks.  Building something physical and programming it to manufacture a product is very detailed work and we have to plan it at the detail level or it will never get done on time or on budget.

Thanks for your prompt and direct response.
